be/src/olap/reader.cpp

```diff
@@ -548,14 +549,36 @@ OLAPStatus Reader::_init_keys_param(const ReaderParams& read_params) {
 
     size_t start_key_size = read_params.start_key.size();
     _keys_param.start_keys.resize(start_key_size, nullptr);
+
+    size_t scan_key_size = read_params.start_key.front().size();
+    if (scan_key_size > _tablet->tablet_schema().num_columns()) {
+        LOG(WARNING)
+                << "Input param are invalid. Column count is bigger than num_columns of schema. "
+                << "column_count=" << scan_key_size
+                << ", schema.num_columns=" << _tablet->tablet_schema().num_columns();
+        return OLAP_ERR_INPUT_PARAMETER_ERROR;
+    }
+
+    std::vector<uint32_t> columns;
+    for (size_t i = 0; i < scan_key_size; ++i) {
+        columns.push_back(i);
+    }
```

Comment:
it is better to extract those to a function

Reply:
- Done. 
- I used the `std::iota` to replace the for-loop.
- Do I need to extra the "if (scan_key_size > _tablet->tablet_schema().num_columns()) { LOG(WARNING)...." to a function, and wrap it in the macro "RETURN_NOT_OK()" ?

Reply:
> it is better to extract those to a function, this is duplicted with line 158

- Done.
- I used the std::iota to replace the for-loop.
- Do I need to extra the "if (scan_key_size > _tablet->tablet_schema().num_columns()) { LOG(WARNING)...." to a function, and wrap it in the macro "RETURN_NOT_OK()" ?